lib/internal/modules/esm/resolve.js

```diff
@@ -136,19 +133,12 @@ function emitLegacyIndexDeprecation(url, packageJSONUrl, base, main) {
 
 const realpathCache = new SafeMap();
 
-/**
- * @param {string | URL} path
- * @returns {import('fs').Stats}
- */
-const tryStatSync =
-  (path) => statSync(path, { throwIfNoEntry: false }) ?? new Stats();
-
 /**
  * @param {string | URL} url
  * @returns {boolean}
  */
 function fileExists(url) {
-  return statSync(url, { throwIfNoEntry: false })?.isFile() ?? false;
+  return internalModuleStat(toPathIfFileURL(url)) === 0;
```

Comment:
On windows the path needs be wrapped with `path.toNamespacedPath()`, which `statSync` does but this doesn't.